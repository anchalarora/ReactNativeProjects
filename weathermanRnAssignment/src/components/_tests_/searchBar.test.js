import React from 'react';
import renderer from 'react-test-renderer';

import SearchBar from '../searchBar';
import {shallow} from 'enzyme';

it('should load search bar view without crashing', () => {
  const searchBarRenderer = renderer.create(<SearchBar />).toJSON();
  expect(searchBarRenderer).toBeTruthy();
});

it('should change state if zipCode is entered', () => {
  const seachBarInstance = renderer
    .create(<SearchBar lable="Search" />)
    .getInstance();
  //console.log(seachBarInstance)
  seachBarInstance.setZipCode('560075');
  expect(seachBarInstance.state.pincode).toEqual('560075');
});

it('should load search bar view without crashing', () => {
  const searchBarComponent = shallow(<SearchBar />);
  expect(searchBarComponent).toMatchSnapshot();
});

it('should load search bar view without crashing', () => {
  const searchBarComponent = renderer.create(<SearchBar />).toJSON();
  expect(searchBarComponent).toMatchSnapshot();
});

let findElement = function (tree, element) {
  let result = undefined;
  for (let i = 0; i < tree.children.length; i++) {
    let childObject = tree.children[i];
    for (let j = 0; j < childObject.children.length; j++) {
      if (childObject.children[j].props.testID == element) {
        result = true;
      }
    }
  }

  return result;

  //   let result = undefined;
  //   let node;
  //   for (node in tree.children) {
  //     if (tree.children[node].children[node].props.testID == element) {
  //       console.log(
  //         'Inside for loop   ',
  //         tree.children[node].children[node].props.testID,
  //       );
  //       result = true;
  //     }
  //   }
  //   return result;
};

it('find TextInput in SearchBar', () => {
  let tree = renderer.create(<SearchBar />).toJSON();
  expect(findElement(tree, 'zipCodeInput')).toBeDefined();
});

it('find Button in SearchBar', () => {
  let tree = renderer.create(<SearchBar />).toJSON();
  expect(findElement(tree, 'searchButton')).toBeDefined();
});
