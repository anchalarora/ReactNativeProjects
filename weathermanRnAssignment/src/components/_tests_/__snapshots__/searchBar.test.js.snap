// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should load search bar view without crashing 1`] = `ShallowWrapper {}`;

exports[`should load search bar view without crashing 2`] = `
<View>
  <View>
    <TextInput
      allowFontScaling={true}
      color="#FFFFFF"
      disableFullscreenUI={true}
      keyboardType="number-pad"
      maxLength={6}
      onChangeText={[Function]}
      placeholder="Enter your pincode"
      placeholderTextColor="#FFFFFF"
      rejectResponderTermination={true}
      style={
        Object {
          "alignItems": "flex-start",
          "borderColor": "#FFFFFF",
          "borderWidth": 1,
          "color": "#14A4F9",
          "fontFamily": "Quicksand-Medium",
          "fontSize": 20,
          "marginTop": 20,
          "padding": 10,
        }
      }
      testID="zipCodeInput"
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      accessibilityRole="button"
      accessibilityState={Object {}}
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="searchButton"
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Search
        </Text>
      </View>
    </View>
  </View>
</View>
`;
